guava/src/com/google/common/escape/CharEscaper.java

```diff
@@ -93,14 +97,22 @@ public String escape(String string) {
    * @return the escaped form of {@code string}
    * @throws NullPointerException if {@code string} is null
    */
-  protected final String escapeSlow(String s, int index) {
+  @SuppressWarnings(value = {"assignment.type.incompatible", "argument.type.incompatible",/*
+  line 132 and line 152: Because of System.arraycopy() method, `rlen` is required to be
+   @LTLengthOf(value={"r", "dest"}, offset={"-1", "destIndex - 1"}).
+   Since rlen = r.length, rlen should already be inferred to have length of @LTLengthOf(value="r", offset="-1" */
+          "compound.assignment.type.incompatible"/*
+          line 147 and line 153 `destIndex` is always @LTEqLengthOf("dest") @LessThan("destSize + 1") because `dest` array
```

Comment:
Referring to the `LessThan` type here isn't that useful to readers, because `destSize` is the length of an array. I would phrase this in terms of upper bound types, instead. That said, it's fairly clear as is, so if you disagree feel free to disregard this comment.